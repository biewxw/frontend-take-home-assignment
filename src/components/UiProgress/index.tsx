import React, { FC } from 'react';
import { Progress, ProgressProps, Flex, Box } from '@chakra-ui/react';

export interface IProgressProps extends ProgressProps {
  leftIcon?: JSX.Element;
  iconColor?: string;
}

const UiProgress: FC<IProgressProps> = ({ leftIcon, iconColor, ...props }) => (
  <Flex alignItems="center" width="100%">
    {leftIcon && (
      <Box marginRight="1rem" color={iconColor}>
        {leftIcon}
      </Box>
    )}
    <Progress width="100%" borderRadius="md" size="sm" {...props} />
  </Flex>
);

export default UiProgress;
