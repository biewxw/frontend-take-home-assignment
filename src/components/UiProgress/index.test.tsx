import React from 'react';
import UiProgress from './index';

import { render } from '@testing-library/react';

describe('<UiProgress />', () => {
  it('should render the normal progressbar', () => {
    const { container } = render(<UiProgress value={20} />);

    expect(container).toMatchSnapshot();
  });

  it('should render the progress with a icon slot', () => {
    const { container } = render(
      <UiProgress value={50} leftIcon={<div>sample</div>} />
    );

    expect(container).toMatchSnapshot();
  });
});
