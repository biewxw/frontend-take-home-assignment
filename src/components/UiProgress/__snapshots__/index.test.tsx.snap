// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UiProgress /> should render the normal progressbar 1`] = `
<div>
  <div
    class="css-7pf6at"
  >
    <div
      class="css-1fkj06j"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="20"
        class="css-18ylzah"
        role="progressbar"
        style="width: 20%;"
      />
    </div>
  </div>
</div>
`;

exports[`<UiProgress /> should render the progress with a icon slot 1`] = `
<div>
  <div
    class="css-7pf6at"
  >
    <div
      class="css-1kr4fmq"
    >
      <div>
        sample
      </div>
    </div>
    <div
      class="css-1fkj06j"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        class="css-18ylzah"
        role="progressbar"
        style="width: 50%;"
      />
    </div>
  </div>
</div>
`;
